python/raydp/spark/dataset.py

```diff
@@ -138,3 +138,18 @@ def __iter__(self):
             tb = reader.read_all()
             df: pd.DataFrame = tb.to_pandas()
             yield df
+
+class RayMLDataset(MLDataset):
+    def __init__(self, df: sql.DataFrame,
+                 num_shards: int,
+                 batch_size: int):
```

Comment:
Is it possible to support parameter fs_directory, compression as well?

Reply:
This is kind of weird to put into `__init__` because `read_parquet` directly returns an `MLDataset`. We can separate it to a static method, and change `__class__` of it to `RayMLDataset` there.